### ISSUE TITLE
Artificial delay added after all code executes, outside of instrument-able code

### ISSUE THREAD

**Issue by zaneclaes** [ChilliCream/graphql-platform#5925]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

I have been attempting to track down why both my HTTP and WebSocket requests are both taking longer than expected to be received by the client (in some cases, a response takes 3+ seconds even though the server code executed in < 100ms). I have integrated the various HC Diagnostic Listeners with DataDog's custom spans, so I am able to see a flame graph of execution. In both HTTP and WebSocket, I see the span fail to close for some time after it finishes, but I cannot instrument any more broadly. 

HTTP example:
- The root span is the request to `aspnet_core`
- The `HTTP HttpPost` span is returned via `ExecuteHttpRequest`
- The top level still takes a LONG time to return (it's only KBs of data)
<img width="935" alt="Screenshot 2023-03-05 at 10 32 14 AM" src="https://user-images.githubusercontent.com/2084821/222976293-8d729839-5552-46eb-9110-eace82318871.png">

At first I thought this might be the HTTP response time due to slow internet connections. But once I started digging into the websocket code, I do not believe that to be true. 

In this case, the `PUB` span is generated by the `OnSubscriptionEvent`, whereas the `SUB` is called by my subscription handler. Notice how the `PUB` span continues to execute long after the `SUB` span finishes...

<img width="1165" alt="Screenshot 2023-03-05 at 10 38 28 AM" src="https://user-images.githubusercontent.com/2084821/222976596-dca151af-19b8-4653-8150-600476925eb2.png">

What's notable is that the log statements at the bottom of the image prints out the JSON response... 

```
  public override IDisposable OnSubscriptionEvent(SubscriptionEventContext context) {
    string name = GetOperationName(context.Subscription);
    Log.Information("[PUB] start {id}:{name}", context.Subscription.Id, name);
    return TraceSpan.Start("PUB", name);
  }

  private string GetOperationName(ISubscription? sub) {
    if (sub?.Operation == null) return "?";
    if (sub.Operation.Name != null) return sub.Operation.Name;
    return sub.Operation.Id;
  }

  public override void SubscriptionEventResult(SubscriptionEventContext context, IQueryResult result) {
    string name = GetOperationName(context.Subscription);
    Log.Information("[SUB] end {id}:{name}", context.Subscription.Id, name);
    base.SubscriptionEventResult(context, result);
  }
```

What's REALLY confusing me is why the `PUB` span is living for so long. According to [the relevant HotChocolate code](https://github.com/ChilliCream/graphql-platform/blob/main/src/HotChocolate/Core/src/Execution/Processing/SubscriptionExecutor.Subscription.cs#L144-L187), the wrapping function should be disposed immediately after `SubscriptionEventResult` is called. 

### Steps to reproduce

See above.

### Additional Context?

The problem gets worse as more data is returned, so perhaps this is HTTP transport time. However, it easily grows to several seconds in duration. Here, we have < 1 MB data returned, yet the `PUB` span lives for > 3 seconds... and the client also sees a 3+ second delay in receiving the payload...

<img width="978" alt="Screenshot 2023-03-05 at 10 46 57 AM" src="https://user-images.githubusercontent.com/2084821/222977034-84ebd6ce-30b0-4e73-9182-7cd35beee1bc.png">

FYI, the many sub-spans you see are Data Loaders executing. But as you can see, they all finish very quickly, and without error.

My assumption is that something is not getting disposed correctly. But I can't imagine what in my code would cause this. I'm not storing anything on the `HttpContext.Items` nor am I using HotChocolate's context/global variables at all. By all accounts, the request should be free to finish and clean up after itself.

### Version

13.0.5

**Comment by PascalSenn** [comment]
@zaneclaes  Can you provide a simple reproduction of this in a plain HC project? Then we can investigate this issue

**Comment by zaneclaes** [comment]
Thanks for taking the time.

Of course, after days of debugging, I finally solved the issue not long after posting ;)

tl;dr — the "uninstrumented" time represents disposing hundreds of objects, as a result of `Transient` scoped objects being used more and more (as I introduced more cases where we use `IResolverContext` to lazy-load).

--- 

I was on the right track with this statement:

> My assumption is that something is not getting disposed correctly.

The WebSocket case was really bothering me, since nothing should have happened between the callback (log statement) and the disposal of the span. Nothing, that is... except garbage collection.

The root cause of the issue was that **my Transient database object grew in complexity over time, and took too long to dispose**. I finally discovered this by putting Log statements in the disposal method:

<img width="1201" alt="Screenshot 2023-03-05 at 3 33 49 PM" src="https://user-images.githubusercontent.com/2084821/223132657-2477db89-cc06-46be-a839-58d1da67ca22.png">

From the black vertical line you can see these log statements happening during the "uninstrumented" time.

This snuck up on me because I was also adding more GraphQL fields that use the `IResolverContext`. It seems that the use of this feature dramatically increases the number of Transient database objects loaded. So the problem was not the response SIZE per se, but the complexity of the Request Graph leading to many Transient scopes being allocated.

The fix was simply to move everything I could from the Transient object into a Scoped object (instead referencing the Scoped from the Transient). By ensuring that all memory allocation, disposal, etc. was on the Scoped object (except for the database itself, which is meant to be Transient for parallelism), I
